### ISSUE TITLE
联合建模日志输出的train-roc_auc异常

### ISSUE THREAD

**Issue by ancongxue** [secretflow/secretflow#1895]
### Issue Type

Correctness/Accuracy

### Installation Source

Binary

### Secretflow Version

secretflow v1.7.0b0

### Operating System and Distribution

Linux x86_64 ,CentOS Linux release 7.1.1503 (Core)

### Python Version

Python 3.10.13

### Issue Description

```shell
现象：
联合建模中，lable为随机生成的0，1。
训练过程中，日志输出“train-roc_auc:0.81565	val-roc_auc:0.50443”，训练结束后，用训练样本进行预测算出的auc为0.52。
日志输出的auc是异常的，最后预测算出的auc比较正常。
导致业务不相信训练的过程是对的。

期待：
日志输出的auc应正确展示本轮训练结束后的auc。

其他：
val-roc_auc是如何算的？训练时只有训练样本入模，测试样本并未入模，这个val auc 代表什么？
```

### Reproduction Code

```shell
sgb_params = get_classic_lightGBM_params()
sgb_params['num_boost_round'] = 200
sgb_params['max_depth'] = 5
sgb_params['learning_rate'] = 0.02
sgb_params['seed'] = 1024
sgb_params['enable_early_stop'] = True
sgb_params['stopping_rounds'] = 200
sgb_params['stopping_tolerance'] = 0.0000000001
sgb_params['enable_monitor'] = True
sgb_params['first_tree_with_label_holder_feature'] = True
sgb_params['save_best_model'] = True
sgb_params['tree_growing_method'] = 'leaf'
pp.pprint(sgb_params)

sgb = Sgb(heu)
sgb_model = sgb.train(sgb_params, train_x, train_y)

# each participant party needs a location to store
sgb_saving_path_dict = {
    device: "./" + device.party + "/uplift"
    for device in train_x.partitions.keys()
}

r = sgb_model.save_model(sgb_saving_path_dict)
wait(r)

# 加载模型
saving_path_dict = {
    mt: './mt/uplift', bank: './bank/uplift'
}
sgb_model = load_model(saving_path_dict, mt)

# 模型预测
yhat = sgb_model.predict(train_x)
yhat = reveal(yhat)
print(yhat)
y = reveal(train_y.partitions[mt].data)
print(y)

from sklearn.metrics import roc_auc_score
print(f"auc: {roc_auc_score(y, yhat)}")
```

### Additional Context or Logs

```shell

```

**Comment by ancongxue** [comment]
以下为日志输出

2025-06-13 06:56:24.248 WARNING callback.py:455 [mt] -- [Anonymous_job] no effective dump_function provided.
2025-06-13 06:56:24.248 INFO level_wise_tree_trainer.py:113 [mt] -- [Anonymous_job] train tree context set up.
2025-06-13 06:56:24.261 INFO level_wise_tree_trainer.py:202 [mt] -- [Anonymous_job] begin train tree.
2025-06-13 06:56:32.635 INFO global_ordermap_booster.py:237 [mt] -- [Anonymous_job] epoch 218 time 8.387020569294691s
2025-06-13 06:56:33.902 INFO callback.py:426 [mt] -- [Anonymous_job] [218]	train-roc_auc:0.81505	val-roc_auc:0.50466

2025-06-13 06:56:33.902 WARNING callback.py:455 [mt] -- [Anonymous_job] no effective dump_function provided.
2025-06-13 06:56:33.902 INFO level_wise_tree_trainer.py:113 [mt] -- [Anonymous_job] train tree context set up.
2025-06-13 06:56:33.915 INFO level_wise_tree_trainer.py:202 [mt] -- [Anonymous_job] begin train tree.
2025-06-13 06:56:42.532 INFO global_ordermap_booster.py:237 [mt] -- [Anonymous_job] epoch 219 time 8.630408216267824s
2025-06-13 06:56:42.825 INFO callback.py:426 [mt] -- [Anonymous_job] [219]	train-roc_auc:0.81565	val-roc_auc:0.50443

**Comment by ancongxue** [comment]
@da-niao-dan 老师，第三点【这意味着日志最后一轮和实际保存的最优模型之间的权重可能差异较大】该怎样去理解呢？权重是指什么？日志最后一轮保存的模型和最有模型之间存在什么权重呢？
